### PR TITLE
Add Batch DVH2CSV

### PR DESCRIPTION
**Pull request by Tamookk**
This PR adds batch processing of exporting DVH to CSV:

- This PR adds another tab to the batch processing window which allows the user to select the output CSV file path for DVH2CSV.
- The functionality performs DVH2CSV on all patients in the selected folder, and saves the DVH data into a single CSV file.
- Checks are made during processing to ensure that a patient dataset includes an RTSTRUCT and an RTDOSE. DVH2CSV is not performed on patients with either of these files missing.
- The batch processing summary screen now includes a summary on DVH2CSV functionality.

This PR also fixes an crash when attempting to import DVH data from an RTDOSE file that does not contain a DVHSequence tag.

### CODE REVIEW

File: src/Controller/BatchProcessingController.py [hunk 1]
```diff
@@ -158,106 +168,155 @@ def perform_processes(self, interrupt_flag, progress_callback):
...
+        Handles creating, starting, and processing the results of batch
+        ISO2ROI.
+        :param interrupt_flag: A threading.Event() object that tells the
+                               function to stop loading.
+        :param progress_callback: A signal that receives the current
+                                  progress of the loading.
+        :param patient: The patient to perform this process on.
+        """
+        # Get current patient files
+        cur_patient_files = self.get_patient_files(patient)
```
Comment (eddieGong2604) [inline src/Controller/BatchProcessingController.py:218]: get_patient_files is a static method so It shouldn't be called by an instance. As a static member holds the state of the class, it's not a good practice that one instance can modify the state of other instances.

consider using your Classname to call get_patient_files

File: src/Controller/BatchProcessingController.py [hunk 2]
```diff
@@ -158,106 +168,155 @@ def perform_processes(self, interrupt_flag, progress_callback):
...
+        Handles creating, starting, and processing the results of batch
+        SUV2ROI.
+        :param interrupt_flag: A threading.Event() object that tells the
+                               function to stop loading.
+        :param progress_callback: A signal that receives the current
+                                  progress of the loading.
+        :param patient: The patient to perform this process on.
+        """
+        # Get patient files
+        cur_patient_files = self.get_patient_files(patient)
```
Comment (eddieGong2604) [inline src/Controller/BatchProcessingController.py:253]: same problem as above

File: src/Controller/BatchProcessingController.py [hunk 3]
```diff
@@ -158,106 +168,155 @@ def perform_processes(self, interrupt_flag, progress_callback):
...
+        Handles creating, starting, and processing the results of batch
+        DVH2CSV.
+        :param interrupt_flag: A threading.Event() object that tells the
+                               function to stop loading.
+        :param progress_callback: A signal that receives the current
+                                  progress of the loading.
+        :param patient: The patient to perform this process on.
+        """
+        # Get current patient files
+        cur_patient_files = self.get_patient_files(patient)
```
Comment (eddieGong2604) [inline src/Controller/BatchProcessingController.py:298]: same problem as above

File: test/test_model_batch_processes.py [hunk 4]
```diff
@@ -136,3 +138,39 @@ def test_batch_suv2roi(test_object):
...
+def test_batch_dvh2csv(test_object):
+    """
+    Test asserts creation of CSV as result of DVH2CSV conversion.
+    :param test_object: test_object function, for accessing the shared
+                        TestObject object.
+    """
+
+    # Loop through patient datasets
+    for patient in test_object.get_patients():
+        cur_patient_files = BatchProcessingController.get_patient_files(
```
Comment (eddieGong2604) [inline test/test_model_batch_processes.py:152]: I think this is the proper way to call get_patient_files

**Review by eddieGong2604 — CHANGES_REQUESTED** [review]
From Team 23: I have tested the DVH2CSV feature individually, and also run it along with ISO2ROI and SUV2ROI and found no problems. The code is relatively clean and well structured. There is one small feedback on the use of the static method. Other than that I'll be happy to approve the PR.

**Comment by Tamookk** [comment]
I see, thank you for the help! I was unaware of this and will fix it up shortly.

File: src/Controller/BatchProcessingController.py (hunk 1 above)
Comment (Tamookk) [reply src/Controller/BatchProcessingController.py]: These have been fixed!

**Review by JThomas2020 — APPROVED** [review]
Team 19, Great job, everything runs smoothly and code is good!

**Review by eddieGong2604 — APPROVED** [review]
From Team 23: Thanks for the fix.

**Pull request merged**